src/pubsub.js

```diff
@@ -9,8 +9,14 @@
     'use strict';
 
     var PubSub = {};
-    root.PubSub = PubSub;
-    factory(PubSub);
+    /* If PubSub event is passed from another app, then root.PubSub will already be present and we can assign that to PubSub variable.
+    If root.PubSub is not present, then we can apply the empty object(var PubSub = {}) to root.PubSub */
+    if (root.PubSub) {
+        PubSub = root.PubSub
```

Comment:
Please add a warning on the console, so the user will see that the library is being loaded more than once and that it might result in problems (as once micro-frontend might expect one version, but getting another).

Something like:

```js
console.warn("PubSub already loaded, using existing version");
```

Reply:
The suggested console.warn statement looks fine to me and have used the same. Thanks :)